src/tests/Common/Coreclr.TestWrapper/MobileAppHandler.cs

```diff
@@ -41,120 +35,119 @@ private static int HandleMobileApp(string action, string platform, string catego
             using (var outputWriter = new StreamWriter(outputStream))
             using (var errorWriter = new StreamWriter(errorStream))
             {
-                if ((platform != "android") && (platform != "apple"))
-                {
-                    outputWriter.WriteLine($"Incorrect value of platform. Provided {platform}. Valid strings are android and apple.");
-                    platformValueFlag = false;
-                }
-
-                if ((action != "install") && (action != "uninstall"))
+                if (File.Exists($"{testBinaryBase}/.retry"))
                 {
-                    outputWriter.WriteLine($"Incorrect value of action. Provided {action}. Valid strings are install and uninstall.");
-                    actionValueFlag = false;
+                    outputWriter.WriteLine("\nretry file was found, quit mobile app installation/uninstallation...");
```

Comment:
We probably still want uninstallation to happen?

Reply:
What benefit would uninstallation bring, when earlier installation didn't succeed? Is there anything to uninstall?

Reply:
Isn't this:
- install
- run
- run
- ...
- crash `   <---- .retry created`
- uninstall `<---- uninstall skipped because .retry exists`

?

Reply:
That `crash` means that there isn't any emulator/app available, right? If so, what is the use of uninstallation?

Reply:
There are more exit codes this happens for other than this crash. If the app itself is somehow bad, there could be cases where we cannot distinguish between infra failures, and we then consider them as infra failures to give the client the benefit of doubt and retry.

I think it's just better to try to uninstall and not leave it behind. If the emulator is really gone, we don't mind much and the uninstallation will fail and we don't care. For all other cases, we're better off removing it.

Reply:
Gotcha! Will add the check.

Reply:
Actually, the checking of `.retry` file for installation/uninstallation should be removed completely, because it only install/uninstall once per test file and different test file will have different `testBinaryBase`, where `.retry` file lives. So the check here only stopped uninstallation before. Now we want uninstallation proceed unconditionally. There is no reason to keep it.